morpholopy/galaxy_data.py

```diff
@@ -433,18 +435,43 @@ def process_galaxy(
     # (re)load the catalogue to read the properties of this particular galaxy
     # we need to disregard the units to avoid problems with the SFR unit
     catalogue = load_catalogue(catalogue_filename, disregard_units=True)
-    groups = load_groups(
-        catalogue_filename.replace(".properties", ".catalog_groups"),
-        catalogue=catalogue,
+    membership_data = h5.File(halo_membership_filename, "r")
+
+    swift_mask = swiftsimio.mask(snapshot_filename, spatial_only=True)
+    # SWIFT data is stored in comoving units, so we need to un-correct
+    # the velociraptor data if it is stored in physical.
+    length_factor = catalogue.a
+
+    halo_x = catalogue.get_quantity("positions.xc")[galaxy_index] / length_factor
+    halo_y = catalogue.get_quantity("positions.yc")[galaxy_index] / length_factor
+    halo_z = catalogue.get_quantity("positions.zc")[galaxy_index] / length_factor
+    r_size = (
+        catalogue.get_quantity("searchradius.search_radius")[galaxy_index]
+        / length_factor
     )
 
-    # get the particles belonging to this galaxy
-    particles, _ = groups.extract_halo(halo_index=galaxy_index)
-    # turn this information into a swiftsimio mask and read the data
-    # using swiftsimio
-    data, mask = to_swiftsimio_dataset(
-        particles, snapshot_filename, generate_extra_mask=True
-    )
+    spatial_mask = [
+        [halo_x - r_size, halo_x + r_size],
+        [halo_y - r_size, halo_y + r_size],
+        [halo_z - r_size, halo_z + r_size],
+    ]
+
+    swift_mask.constrain_spatial(spatial_mask)
```

Comment:
This spatial mask is intended to be used. However, currently I cannot use it because otherwise, I cannot match particle IDs from the snapshot and the soap membership file.